src/main/java/org/spongepowered/common/data/manipulator/mutable/entity/SpongeBodyPartRotationalData.java

```diff
@@ -0,0 +1,166 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.manipulator.mutable.entity;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.flowpowered.math.vector.Vector3d;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Maps;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.MemoryDataContainer;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.entity.ImmutableBodyPartRotationalData;
+import org.spongepowered.api.data.manipulator.mutable.entity.BodyPartRotationalData;
+import org.spongepowered.api.data.type.BodyPart;
+import org.spongepowered.api.data.type.BodyParts;
+import org.spongepowered.api.data.value.mutable.MapValue;
+import org.spongepowered.api.data.value.mutable.Value;
+import org.spongepowered.common.data.manipulator.immutable.entity.ImmutableSpongeBodyPartRotationalData;
+import org.spongepowered.common.data.manipulator.mutable.common.AbstractData;
+import org.spongepowered.common.data.util.DataConstants;
+import org.spongepowered.common.data.util.ImplementationRequiredForTest;
+import org.spongepowered.common.data.value.mutable.SpongeMapValue;
+import org.spongepowered.common.data.value.mutable.SpongeValue;
+
+import java.util.Map;
+
+@ImplementationRequiredForTest
+public class SpongeBodyPartRotationalData extends AbstractData<BodyPartRotationalData, ImmutableBodyPartRotationalData> implements BodyPartRotationalData {
+
+    private Map<BodyPart, Vector3d> rotations;
+
+    public SpongeBodyPartRotationalData() {
+        this(ImmutableMap.<BodyPart, Vector3d>builder()
+                .put(BodyParts.HEAD, DataConstants.DEFAULT_HEAD_ROTATION)
+                .put(BodyParts.CHEST, DataConstants.DEFAULT_CHEST_ROTATION)
+                .put(BodyParts.LEFT_ARM, DataConstants.DEFAULT_LEFT_ARM_ROTATION)
+                .put(BodyParts.RIGHT_ARM, DataConstants.DEFAULT_RIGHT_ARM_ROTATION)
+                .put(BodyParts.LEFT_LEG, DataConstants.DEFAULT_LEFT_LEG_ROTATION)
+                .put(BodyParts.RIGHT_LEG, DataConstants.DEFAULT_RIGHT_LEG_ROTATION)
+                .build()
+        );
+    }
+
+    public SpongeBodyPartRotationalData(Map<BodyPart, Vector3d> rotations) {
+        super(BodyPartRotationalData.class);
+        this.rotations = Maps.newHashMap(checkNotNull(rotations, "rotations"));
+        registerGettersAndSetters();
+    }
+
+    @Override
+    public int compareTo(BodyPartRotationalData o) {
+        return 0;
```

Comment:
Need to implement the comparable. Since each value is comparable. Granted the comparison will rely on the map keys having the same keys, then comparing each key.